src/shared/mount-util.c

```diff
@@ -1813,3 +1813,13 @@ int make_fsmount(
 
         return TAKE_FD(mnt_fd);
 }
+
+char *umount_and_unlink_and_free(char *p) {
+        PROTECT_ERRNO;
+        if (!p)
+                return NULL;
+
+        (void) umount2(p, 0);
+        (void) unlink(p);
+        return mfree(p);
+}
```

Comment:
Hmm, the other two helpers are static inline. This one is not any bigger, so make things consistent?

Reply:
I couldnt make that work, because that would mean we'd have to include sys/mount.h in the header file but that breaks the build on some older distros because some definitions conflict with linux/fs.h and you hence have to choose. Hence not easily doable. But i think the function is long enough that its probably not worth to prematurely optimize here anyway, both the regular and the static inline version is probably fine. And LTO probably ensures that both versions would result in the same code anyway...